### PR TITLE
Fixes for GUI2

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
Addresses several issues found during book club.
- Add placeholder icons for visualizations (see screenshot)
  - Also add the corresponding export to visualizations, and relevant support in the visualization store and visualization metadata DB.
  - Show icon both on the visualization selector button, and in each visualization selector entry
  - Adjust gaps between visualization selector entries
- Port table viz fixes (#8102) to Vue
- Fix height of table rows to avoid cutting off descenders on letters like `y` and `g`
- Make the space and enter keys work on visualization toolbar buttons, if they are selected
- `.blur()` code editor when clicking outside of it
  - And similarly `.blur()` visualization selector
- Move selection brush on scroll by scaling the delta of `scrollTop` and `scrollLeft`
  - Note that mouse position is technically still incorrect when scrolling past the end. I think this is fine - the new behavior is less broken, plus I am not aware of any way to fix this.

### Important Notes
None

### Screenshots
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by somebody1234** [comment]
small issue: the new blur handler breaks the CB, which gets blurred before a suggestion is accepted

**Comment by somebody1234** [comment]
the merge with main was not required, but is useful to test with the fixes for CB. i think dropping a node from an edge is broken though?

**Comment by farmaazon** [comment]
> i think dropping a node from an edge is broken though?

[It's not implemented yet](8054)

**Comment by somebody1234** [comment]
right - bad wording on my part

File: app/gui2/src/stores/visualization/compilerMessaging.ts [hunk 1]
```diff
@@ -34,10 +36,19 @@ export const stylePathAttribute = 'data-style-path'
 
 const VisualizationModule = z.object({
   // This is UNSAFE, but unavoiable as the type of `Visualization` is too difficult to statically
-  // check. Insteead it will be caught by Vue when trying to mount the visualization, and replaced
+  // check. Instead it will be caught by Vue when trying to mount the visualization, and replaced
   // with a 'Loading Error' visualization.
   default: z.custom<Visualization>(() => true),
   name: z.string(),
+  // The name of an icon, or a URL or data URL. If it contains `:`, it is assumed to be a URL.
```
Comment (Frizi) [inline app/gui2/src/stores/visualization/compilerMessaging.ts:45]: This URL detection logic seems to not be implemented.

**Review by Frizi — APPROVED** [review]

File: app/gui2/src/stores/visualization/compilerMessaging.ts (hunk 1 above)
Comment (somebody1234) [reply app/gui2/src/stores/visualization/compilerMessaging.ts:45]: ah... nice catch, thanks :sweat_smile:

**Pull request merged**